### PR TITLE
fib and difference

### PR DESCRIPTION
**Pull request by AleksandrHorev**

### CODE REVIEW

File: Fibonache.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// with Recursion. it works once, need to reset resultFibonachhiRecursion each time 
+function fibonachhiRecursion(countOfNumbers) {
```
Comment (ktttnv) [inline Fibonache.js]: Function should return fibonacci number for the given index

File: Fibonache.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  if (countOfNumbers < 2) throw new Error("countOfNumbers should be at least 2");
```
Comment (ktttnv) [inline Fibonache.js]: We know 0th and 1st fibonacci numbers so it's okay to have fibonacci numbers' array length to be equal to 1 or 2

File: differenceMinMaxValue.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+function differenceMinMaxValue(array) {
+  if (array.lenght < 1) {
+    throw new Error("Array must contain at least 2 numbers");
```
Comment (ktttnv) [inline differenceMinMaxValue.js]: It's okay to have an array with the length of 1

File: differenceMinMaxValue.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  let max = array[0];
+  let min = array[1];
```
Comment (ktttnv) [inline differenceMinMaxValue.js]: In should be changed at least for the case of array length = 1. **Hint:** it's better (easier) to initialize the maximum and minimum to `array[0]`.

File: 1/numberDivisors.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// complexity is O(log n) - I'm not sure
+function numberDivisors(number) {
+  const result = [];
+  for (let divisor = 1; divisor < (number/divisor + 1); divisor++) {
+    if (number % divisor === 0 && !result[divisor]) { result.push(divisor); result.push(number/divisor);}
```
Comment (ktttnv) [inline 1/numberDivisors.js]: What does `!result[divisor]` condition check? Why do we need it here?

File: 1/numberDivisors.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  }
+  return result;
+}
+
+
+
+
+console.log(numberDivisors(6)); // [1,6,2,3]
+console.log(numberDivisors(1)); // [1, 1] a side effect for 1
```
Comment (ktttnv) [inline 1/numberDivisors.js:16]: All numbers that are squares will have this effect.
_For example:_ for input 36, the result will be: [1, 36, 2, 18, 3, 12, 4, 9, 6, 6].
There should be no duplicate numbers.

File: 1/numberDivisors.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+// complexity is O(log n) - I'm not sure
```
Comment (ktttnv) [inline 1/numberDivisors.js:1]: The complexity of this algorithm is `O(sqrt(n))`. We'll discuss why

File: 1/numberDivisors.js (hunk 5 above)
Comment (AleksandrHorev) [reply 1/numberDivisors.js]: ~~example:
the divisor is `1`, `number/divisor` is `X`. I put in the `result` 1 and X. When we face `X` as a divisor we shouldn't add it again~~
got it, there will not be a variant like `divisor = X`

File: 1/numberDivisors.js (hunk 6 above)
Comment (AleksandrHorev) [reply 1/numberDivisors.js:16]: thnks, fixed

File: 1/numberDivisors.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+      divisor === number/divisor? {} : result.push(number/divisor);
+    }
+  }
+  return result;
+}
+
+
+console.log(numberDivisors(6)); // [1,6,2,3]
+console.log(numberDivisors(49)); // [ 1, 49, 7, 7]
+console.log(numberDivisors(1)); // [1, 1] a side effect for 1
```
Comment (ktttnv) [inline 1/numberDivisors.js:16]:
```suggestion
console.log(numberDivisors(49)); // [ 1, 49, 7]
console.log(numberDivisors(1)); // [ 1 ]
```

I guess it works properly now :)

**Review by ktttnv — APPROVED** [review]

**Pull request merged**